homeassistant/components/device_tracker/icloud.py

```diff
@@ -19,69 +19,126 @@
-        _LOGGER.error('Must specify a username and password')
-        return False
+    iclouddevice = Icloud(hass, username, password, name)
+    ICLOUDTRACKERS[name] = iclouddevice
 
-    try:
-        _LOGGER.info('Logging into iCloud Account')
-        # Attempt the login to iCloud
-        api = PyiCloudService(username,
-                              password,
-                              verify=True)
-    except PyiCloudFailedLoginException as error:
-        _LOGGER.exception('Error logging into iCloud Service: %s', error)
-        return False
+    def lost_iphone(call):
+        """ Calls the lost iphone function if the device is found """
+        accountname = call.data.get('accountname')
+        devicename = call.data.get('devicename')
+        if accountname in ICLOUDTRACKERS:
+            ICLOUDTRACKERS[accountname].lost_iphone(devicename)
 
-    def keep_alive(now):
-        """ Keeps authenticating iCloud connection. """
-        api.authenticate()
-        _LOGGER.info("Authenticate against iCloud")
+    hass.services.register('device_tracker', 'lost_iphone',
+                           lost_iphone)
 
-    track_utc_time_change(hass, keep_alive, second=0)
+    def update_icloud(call):
+        """ Calls the update function of an icloud account """
+        accountname = call.data.get('accountname')
+        if accountname in ICLOUDTRACKERS:
+            ICLOUDTRACKERS[accountname].update_icloud(see)
 
-    def update_icloud(now):
-        """ Authenticate against iCloud and scan for devices. """
-        try:
-            # The session timeouts if we are not using it so we
-            # have to re-authenticate. This will send an email.
-            api.authenticate()
-            # Loop through every device registered with the iCloud account
-            for device in api.devices:
-                status = device.status()
-                location = device.location()
-                # If the device has a location add it. If not do nothing
-                if location:
-                    see(
-                        dev_id=re.sub(r"(\s|\W|')",
-                                      '',
-                                      status['name']),
-                        host_name=status['name'],
-                        gps=(location['latitude'], location['longitude']),
-                        battery=status['batteryLevel']*100,
-                        gps_accuracy=location['horizontalAccuracy']
-                    )
-                else:
-                    # No location found for the device so continue
-                    continue
-        except PyiCloudNoDevicesException:
-            _LOGGER.info('No iCloud Devices found!')
+    hass.services.register('device_tracker',
+                           'update_icloud', update_icloud)
+    iclouddevice.update_icloud(see)
 
     track_utc_time_change(
-        hass, update_icloud,
+        hass, iclouddevice.update_icloud(see),
```

Comment:
hmm, the only thing I see in my log around the time it should try to poll is the following error:
```
16-02-17 09:50:00 homeassistant.core: BusHandler:Exception doing job
Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\homeassistant-0.14.0.dev0-py3.4.egg\homeassistant\core.py", line 789, in job_handler
    func(arg)
  File "C:\Python34\lib\site-packages\homeassistant-0.14.0.dev0-py3.4.egg\homeassistant\helpers\event.py", line 186, in pattern_time_change_listener
    action(now)
TypeError: 'NoneType' object is not callable
```
Could this the error you mean?